### PR TITLE
Update `scripts/update_base_image.py` to work on workflows

### LINKED ISSUE #29: Modify update script to update matrix
Now that there is a matrix from #28, we should modify the [update script](https://github.com/mamba-org/micromamba-devcontainer/blob/main/scripts/update_base_image.py) so that the [matrix](https://github.com/mamba-org/micromamba-devcontainer/blob/b34ef0368aab25f63e6efa35335af0605f2ceba5/.github/workflows/publish_image.yml#L21-L22) is kept up-to-date.

CC @basnijholt

### PR DESCRIPTION
**Pull request by basnijholt**
Closes #29

### CODE REVIEW

File: scripts/update_base_image.py [hunk 1]
```diff
@@ -13,74 +18,160 @@
...
+    with WORKFLOW_PATH.open("r") as file:
+        lines = file.readlines()
+
+    existing_base_images = []
+    matrix_start = None
+    for i, line in enumerate(lines):
+        if "base-image:" in line:
+            matrix_start = i + 1
+        elif matrix_start is not None and line.strip().startswith("-"):
+            image = line.strip()[2:].strip()
```
Comment (maresb) [inline scripts/update_base_image.py]: Why not this?

```suggestion
            image = line.strip()[1:].strip()
```

File: scripts/update_base_image.py [hunk 2]
```diff
@@ -13,74 +18,160 @@
...
-    print(f"Base image digest: {digest}")
+    # Find the end of the matrix section
+    matrix_end = matrix_start
+    for i in range(matrix_start, len(lines)):
+        if not lines[i].strip().startswith("-"):
+            break
+        matrix_end = i
 
+    # Replace the matrix entries with new base images
+    matrix_lines = [f"          - {image}\n" for image in new_base_images]
```
Comment (maresb) [inline scripts/update_base_image.py]:
```suggestion
    indentation_size = len(lines[matrix_end]) - len(lines[matrix_end].lstrip())
    indentation_str = indentation * " "

    # Replace the matrix entries with new base images
    matrix_lines = [indentation_str + f"- {image}\n" for image in new_base_images]
```

**Review by maresb — APPROVED** [review]
Awesome, thanks so much for the rewrite!!! Just a few very minor suggestions...

File: scripts/update_base_image.py (hunk 1 above)
Comment (basnijholt) [reply scripts/update_base_image.py]: That works too!

**Comment by basnijholt** [comment]
Thanks for your review, I applied your change 😄

**Pull request merged**